File: src/main/java/cs/android/view/list/CSListAdapter.java
package cs.android.view.list;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

public class CSListAdapter extends BaseAdapter {

    private final CSListController _controller;

    public CSListAdapter(CSListController controller) {
        _controller = controller;
    }

    public int getCount() {
        return _controller.size();
    }

    public int getViewTypeCount() {
        return _controller.getItemTypesCount();
    }

    public boolean isEnabled(int position) {
        return _controller.isEnabled(position);
    }

    public Object getItem(int position) {
        return _controller.dataAt(position);
    }

    public int getItemViewType(int position) {
        return _controller.getItemViewType(position);
    }

    public long getItemId(int position) {
        return position;
    }

    public View getView(int position, View view, ViewGroup parent) {
        return _controller.getRowView(position, view);
    }
}

File: library/renetik-controller-listview/src/main/java/renetik/android/listview/CSListAdapter.kt
package renetik.android.listview

import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import renetik.android.java.extensions.collections.at

class CSListAdapter(private val controller: CSListController<*, *>) : BaseAdapter() {

    override fun getCount() =
        controller.dataCount

    override fun getViewTypeCount() =
        controller.viewTypesCount

    override fun isEnabled(position: Int) =
        controller.isEnabled(position)

    override fun getItem(position: Int) =
        controller.dataAt(position)

    override fun getItemViewType(position: Int) =
        controller.getItemViewType(position)

    override fun getItemId(position: Int) =
        position.toLong()

    override fun getView(position: Int, view: View?, parent: ViewGroup?) =
            controller.getRowView(position, view)
}